python_modules/dagster/dagster/_streamline/asset_materialization_health.py

```diff
@@ -69,6 +73,20 @@ async def compute_for_asset(
             if materialized_partition_subset is None or failed_partition_subset is None:
                 check.failed("Expected partitions subset for a partitioned asset")
 
+            last_run_id = None
+            if asset_record is not None:
+                storage_id_to_run_id = {
+                    asset_record.asset_entry.last_materialization_storage_id: asset_record.asset_entry.last_materialization_record.run_id
+                    if asset_record.asset_entry.last_materialization_record
+                    else None,
+                    asset_record.asset_entry.last_failed_to_materialize_storage_id: asset_record.asset_entry.last_failed_to_materialize_record.run_id
+                    if asset_record.asset_entry.last_failed_to_materialize_record
+                    else None,
+                }
+                max_storage_id = max(key for key in storage_id_to_run_id.keys() if key is not None)
```

Comment:
I think this will error if you call this on an asset record that does not have either of these values set (i.e. `max(<empty sequence>)` will result in an error)

An alt pattern:

```python
entry = asset_record.asset_entry
latest_record = max(
    [
        entry.last_materialization_record,
        entry.last_failed_to_materialize_record,
    ],
    key=lambda record: -1 if record is None else record.storage_id,
)
last_run_id = latest_record.storage_id if latest_record else None
```

Reply:
ah good call, i didn't know that. will fix